nixos/modules/system/boot/initrd-network.nix

```diff
@@ -144,13 +144,15 @@ in
         # Acquire DHCP leases.
         for iface in ${dhcpIfShellExpr}; do
           echo "acquiring IP address via DHCP on $iface..."
-          udhcpc --quit --now -i $iface -O staticroutes --script ${udhcpcScript} ${udhcpcArgs}
+          # run DHCP in the background until it acquires an address
+          udhcpc --quit -i $iface -O staticroutes --script ${udhcpcScript} ${udhcpcArgs} &
         done
       ''
 
       + cfg.postCommands));
 
     boot.initrd.postMountCommands = mkIf (cfg.flushBeforeStage2 && !config.boot.initrd.systemd.enable) ''
+      pkill udhcpc
```

Comment:
Thanks for the comment! I think pkill will only match "udhcpc" exactly here (it needs the -f option to match substrings) but let me look into possibly collecting the PIDs and storing them in a variable. It would be more elegant.

Reply:
I was wrong: pill needs --exact and then it will require an exact match on the process name. I'll still look at collecting the PIDs though.

Reply:
I don't think this pkill is even needed. The interfaces will all be taken down before initrd ends. If udhcpc is still running, it will exit when the underlying interface goes away.